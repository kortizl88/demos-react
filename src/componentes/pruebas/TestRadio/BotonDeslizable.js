import React from 'react'

export const BotonDeslizable = ({ leyenda, inactivo = false, estado,  eventoCambioEstado }) => {
    return (
        <>
            <label>{leyenda}</label>
            <input 
                type='checkbox'
                checked={estado}
                disabled={inactivo}
                onChange={(e) => eventoCambioEstado(e.target.checked)}
            />
        </>
    )
}
