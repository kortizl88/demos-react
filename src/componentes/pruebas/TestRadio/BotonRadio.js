import React from 'react'

export const BotonRadio = ({leyenda, inactivo=false, valor, estado, eventoCambioEstado}) => {
    return (
        <>
            <label>{leyenda}</label>
            <input type="radio"
                    checked={ estado === valor }
                    disabled={inactivo}
                    value={valor}
                    onChange={(e) => eventoCambioEstado(e.target.value)}
                    />  
        </>
    )
}
