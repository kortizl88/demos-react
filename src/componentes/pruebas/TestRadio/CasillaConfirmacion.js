import React from 'react'
import './CasillaConfirmacion.scss';

export const CasillaConfirmacion = ({estado, eventoCambio, identificador, texto, inactivo=false }) => {
    
    const estiloEtiqueta = () => {
        if(inactivo){
            return 'etiqueta inactivo'
        }else{
            return 'etiqueta';
        }
    };

    return (
        <div className='casilla-confirmacion'>
            <label className={estiloEtiqueta()} htmlFor={identificador} >{texto}
                <input 
                    className='control' 
                    type="checkbox" 
                    id={identificador}
                    disabled={inactivo}
                    checked={estado}
                    onChange={()=> eventoCambio(!estado)}/>
                <span className="indicador"/>
            </label>
        </div>
    )
}
