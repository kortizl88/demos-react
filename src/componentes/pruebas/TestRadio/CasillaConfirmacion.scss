.casilla-confirmacion{
    .etiqueta{
        position: relative;
        cursor: pointer;
        font-size: 14px;
        line-height: 14px;
        padding-left: 20px;
        &.inactivo{
            color:#CCCDCE;
        }
        input.control {
            position: absolute;
            opacity: 0;
            cursor: pointer;
            height: 0;
            width: 0;
          }
        .indicador{
            position: absolute;
            top: 1px;
            left: 0;
            height: 12px;
            width: 12px;
            margin-right: 8px;
            background-color: #eee;
            border: 1px solid black;
            &:after {
                content: "";
                position: absolute;
                display: none;
              }
        }

        input.control:checked ~ .indicador {
            background-color: #006341;
        }

        input.control:checked ~ .indicador:after {
            display: block;
        }

        .indicador:after {
            left: 3px;
            top: 0px;
            width: 4px;
            height: 8px;
            border: solid white;
            border-width: 0 2px 2px 0;
            transform: rotate(35deg);
          }

        input.control:disabled ~ .indicador {
            background-color: #fff;
            border-color:#CCCDCE;
        }

    }
}