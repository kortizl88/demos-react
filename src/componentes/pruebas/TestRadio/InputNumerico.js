import React, {useState, useEffect} from 'react'
import NumberFormat from 'react-number-format';

export const InputNumerico = () => {
    const totalVenta = 7867;
    const [enganche, setEnganche] = useState({
        monto: 1500,
        porcentaje: parseInt(( 1500 / totalVenta ) * 100)
    })

    const calcular = (e, tipo) => {
        if(e.key === 'Enter'){
            let valor = e.target.value.toString();
            valor = Number(valor.replace(/[^0-9.-]/g,""))
            if(tipo === 1){
                setEnganche({
                    monto: valor,
                    porcentaje: parseInt(( valor / totalVenta ) * 100 )
                })
            }else{
                setEnganche({
                    monto: parseInt(( totalVenta * valor ) / 100) ,
                    porcentaje : valor
                })
            }
        }
    }

    return (
        <div>
            <p>Numero</p>
            <NumberFormat
                className='monto' 
                value={enganche.monto}
                onChange={e => setEnganche( { ...enganche,  monto: e.target.value } ) }
                onKeyPress={(e) => calcular(e, 1)}
                thousandSeparator={true}
                prefix={'$'} />
            <p>Porcentaje</p>
            <NumberFormat
                className='porcentaje'
                value={enganche.porcentaje}
                onChange={ e => setEnganche( { ...enganche, porcentaje: e.target.value  } ) }
                onKeyPress={(e) => calcular(e, 2)}
                suffix={'%'} />            
        </div>
    )
}
