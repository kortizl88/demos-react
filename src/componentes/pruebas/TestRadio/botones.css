.control {
    font-family: arial;
    display: block;
    position: relative;
    padding-left: 30px;
    margin-bottom: 0px;
    padding-top: 0px;
    cursor: pointer;
    font-size: 16px;
}
    .control input {
        position: absolute;
        z-index: -1;
        opacity: 0;
    }
.control_indicator {
    position: absolute;
    top: 1px;
    left: 0;
    height: 20px;
    width: 20px;
    background: #ffffff;
    border: 1px solid #266b19;
    border-radius: undefinedpx;
}
.control:hover input ~ .control_indicator,
.control input:focus ~ .control_indicator {
    background: #ffffff;
}

.control input:checked ~ .control_indicator {
    background: #ffffff;
}
.control:hover input:not([disabled]):checked ~ .control_indicator,
.control input:checked:focus ~ .control_indicator {
    background: #ffffff;
}
.control input:disabled ~ .control_indicator {
    background: #e6e6e6;
    opacity: 1;
    pointer-events: none;
}
.control_indicator:after {
    box-sizing: unset;
    content: '';
    position: absolute;
    display: none;
}
.control input:checked ~ .control_indicator:after {
    display: block;
}
.control-radio .control_indicator {
    border-radius: 50%;
}

.control-radio .control_indicator:after {
    left: 2px;
    top: 2px;
    height: 14px;
    width: 14px;
    border-radius: 50%;
    background: #38662f;
    transition: background 250ms;
}
.control-radio input:disabled ~ .control_indicator:after {
    background: #7b7b7b;
}