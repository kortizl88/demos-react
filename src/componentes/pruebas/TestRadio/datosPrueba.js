export const menuCat = [
    {
        id: 1,
        descripcion: 'TV',
        idPadre: 0,
        nivel: 1,
    },
    {
        id: 2,
        descripcion: 'LB',
        idPadre: 0,
        nivel: 1,
    },
    {
        id: 3,
        descripcion: 'OTRO',
        idPadre: 0,
        nivel: 1,
    },
    {
        id: 4,
        descripcion: 'ETC',
        idPadre: 0,
        nivel: 1,
    }
]