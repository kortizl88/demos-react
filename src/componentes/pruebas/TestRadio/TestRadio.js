import React, {useState} from 'react'
import { BotonDeslizable } from './BotonDeslizable';
import { BotonRadio } from './BotonRadio';
import './botones.css'

export const TestRadio = () => {

    
    const [isChecked, setIsChecked] = useState(false);
    const [radio, setRadio] = useState();

    return (
        <div>
            <h1>Pruebas de radios con useState</h1>

            <h2>Check esta checado? : { isChecked ? 'true' : 'false' } </h2>

            <h2>Radio seleccionado : { radio } </h2>

            <form>
                <h3>Check</h3>
                <BotonDeslizable 
                    leyenda='mi check personalizado'
                    estado={isChecked}
                    eventoCambioEstado={setIsChecked}
                    />

                <br/>
                <h3>Radios</h3>
                <BotonRadio 
                    leyenda="Manzana"
                    valor='manzana'
                    estado={radio}
                    eventoCambioEstado={setRadio} />
                <br/>
                <BotonRadio 
                    leyenda="Pera"
                    valor='pera'
                    estado={radio}
                    eventoCambioEstado={setRadio} />
                <br/>
                <BotonRadio 
                    leyenda="Platano"
                    valor='platano'
                    estado={radio}
                    eventoCambioEstado={setRadio} />

                    <br/>
                    <br/>
                    <br/>
                    <div class="control-group">
                        <label class="control control-radio">
                            First radio
                            <input type="radio" name="radio" checked="checked" />
                            <div class="control_indicator"></div>
                        </label>
                        <label class="control control-radio">
                            Second radio
                            <input type="radio" name="radio" />
                            <div class="control_indicator"></div>
                        </label>
                        <label class="control control-radio">
                            Disabled
                            <input type="radio" name="radio2" disabled="disabled" />
                            <div class="control_indicator"></div>
                        </label>
                        <label class="control control-radio">
                            Disabled & checked
                            <input type="radio" name="radio2" disabled="disabled" checked="checked" />
                            <div class="control_indicator"></div>
                        </label>
                    </div>

            </form>
        </div>
    )
}
