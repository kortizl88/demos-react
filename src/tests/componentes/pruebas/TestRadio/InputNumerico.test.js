import { mount } from "enzyme"
import { InputNumerico } from "../../../../componentes/pruebas/TestRadio/InputNumerico";
import {act} from '@testing-library/react'

describe('Prueba en InputNumerico', () => {
    
    const componente = mount(<InputNumerico />);

    test('Debe mostrarse correctamente', () => {
        expect(componente).toMatchSnapshot();
        expect(componente.find('NumberFormat.monto').instance().props.value).toBe(1500);
        expect(componente.find('NumberFormat.monto').instance().props.thousandSeparator).toBe(true);
        expect(componente.find('NumberFormat.monto').instance().props.prefix).toBe('$');

        expect(componente.find('NumberFormat.porcentaje').instance().props.value).toBe(19);
        expect(componente.find('NumberFormat.porcentaje').instance().props.suffix).toBe('%');
    });

    test('Debe actualizar el valor del input monto', () => {
        act(() => {
            componente.find('NumberFormat.monto').instance().props.onChange(
                 { target: { value: 4000 } });
        });
        expect(componente.find('NumberFormat.monto').instance().props.value).toBe(4000);
    });

    test('Debe actualizar el valor del input porcentaje', () => {
        act(() => {
            componente.find('NumberFormat.porcentaje').instance().props.onChange(
                 { target: { value: 50 } });
        });
        expect(componente.find('NumberFormat.porcentaje').instance().props.value).toBe(50);
    });
    

    test('Debe cambiar el valor del porcentaje cuando cambie el monto', () => {
        act(() => {
            componente.find('NumberFormat.monto').instance().props.onKeyPress(
                 { key: "Enter", target: { value: '$5,000' } });
        });
        expect(componente.find('NumberFormat.monto').instance().props.value).toBe(5000);
        expect(componente.find('NumberFormat.porcentaje').instance().props.value).toBe(63);
    })

    test('Debe cambiar el valor del monto cuando cambie el porcentaje', () => {
        act(() => {
            componente.find('NumberFormat.porcentaje').instance().props.onKeyPress(
                 { key: "Enter", target: { value: '50%' } });
        });
        expect(componente.find('NumberFormat.monto').instance().props.value).toBe(3933);
        expect(componente.find('NumberFormat.porcentaje').instance().props.value).toBe(50);
    })

    test('NO Debe cambiar el valor del monto cuando cambie el porcentaje', () => {
        act(() => {
            componente.find('NumberFormat.porcentaje').instance().props.onKeyPress(
                 { key: "Otro", target: { value: '$10' } });
        });
        expect(componente.find('NumberFormat.monto').instance().props.value).toBe(3933);
        expect(componente.find('NumberFormat.porcentaje').instance().props.value).toBe(50);
    })
    
    
})
