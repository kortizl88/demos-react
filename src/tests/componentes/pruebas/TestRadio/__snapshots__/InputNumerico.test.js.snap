// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en InputNumerico Debe mostrarse correctamente 1`] = `ReactWrapper {}`;
